src/group_theory/subgroup.lean

```diff
@@ -1662,6 +1714,147 @@ set.subset.antisymm
     by { rw sup_eq_closure, apply Inf_le _, dsimp, refl })
   ((sup_eq_closure N H).symm ▸ subset_closure)
 
-end subgroup
+@[to_additive] lemma mem_mul_iff
+  {H K : subgroup G} {g : G} (h : ↑(H ⊔ K) = (H : set G) * K) :
+  g ∈ H ⊔ K ↔ ∃ x y, x ∈ H ∧ y ∈ K ∧ x * y = g :=
+begin
+  refine ⟨λ (hg : g ∈ ↑(H ⊔ K)), _, λ hg,  (_ : g ∈ ↑(H ⊔ K))⟩,
+  rwa [h, set.mem_mul] at hg,
+  rwa [h, set.mem_mul],
+end
```

Comment:
This lemma seems a bit suspect to me, given it can be proved as:
```suggestion
iff.trans (by exact set.ext_iff.1 h g) set.mem_mul
```
The name `mem_mul_iff` is also odd, as it's actually a lemma about membership of `H ⊔ K`